### PR TITLE
Feature/#130 Validate release-droid config

### LINKED ISSUE #130: Add validation for release_config.yml
Release droid >= 0.8.0 requires a release_config.yml file.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #130

### CODE REVIEW

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/FileTemplatesFactory.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
     List<FileTemplate> getGlobalTemplates(final List<AnalyzedSource> sources) {
         final List<FileTemplate> templates = new ArrayList<>();
         templates.add(new FileTemplateFromResource(".github/workflows/broken_links_checker.yml", REQUIRE_EXACT));
+        templates.add(new FileTemplateFromResource("release_config.yml", REQUIRE_EXIST));
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/FileTemplatesFactory.java:27]: ✅

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/RequiredFileValidator.java [hunk 2]
```diff
@@ -77,6 +79,29 @@ private static String unifyNewlines(final String content) {
...
+                    .of(SimpleValidationFinding
+                            .withMessage(ExaError.messageBuilder("E-PK-CORE-119")
+                                    .message("Missing required file: '{{required file}}'")
+                                    .parameter("required file", projectDir.relativize(file)).toString())
+                            .andFix(fix).build());
+        } else {
+            return emptyList();
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/RequiredFileValidator.java:104]: ✅

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/ProjectFilesValidator.java [hunk 3]
```diff
@@ -74,10 +75,29 @@ private Path directoryOf(final Path path) {
...
+            final Path projectFile) {
+        return new RequiredFileValidator().validateFileExists(projectDirectory, projectFile, template.getContent());
+    }
+
+    private List<ValidationFinding> validateExactFileContent(final Path projectDirectory, final FileTemplate template,
+            final Path projectFile) {
         final String templateContent = template.getContent();
-        return new RequiredFileValidator().validateFile(relativeDirectory, projectFile,
+        return new RequiredFileValidator().validateFile(projectDirectory, projectFile,
                 withContentEqualTo(templateContent));
```
[the remark this answers is not in the document]
Comment (kaklakariada) [reply project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/ProjectFilesValidator.java]: 👍

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/ProjectFilesValidator.java [hunk 4]
```diff
@@ -74,10 +77,22 @@ private Path directoryOf(final Path path) {
...
+        switch (template.getTemplateType()) {
+        case REQUIRE_EXACT:
+            return withContentEqualTo(template.getContent());
+        case REQUIRE_EXIST:
+            return fileExists(template.getContent());
+        default:
+            throw new IllegalStateException(ExaError.messageBuilder("E-PK-CORE-119")
+                    .message("Unknown template type {{template type}}", template.getTemplateType()).ticketMitigation()
+                    .toString());
+        }
```
Comment (jakobbraun) [inline project-keeper/src/main/java/com/exasol/projectkeeper/validators/files/ProjectFilesValidator.java:96]: :+1: lloks nice now

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**